src/module.c

```diff
@@ -4067,21 +4082,35 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         }
     }
 
-    /* If we are using single commands replication, we need to wrap what
-     * we propagate into a MULTI/EXEC block, so that it will be atomic like
-     * a Lua script in the context of AOF and slaves. */
-    if (replicate) moduleReplicateMultiIfNeeded(ctx);
-
-    /* Run the command */
+    /* Handle command flags and replication */
+    int reset_replication_allowed = 0;
     int call_flags = CMD_CALL_SLOWLOG | CMD_CALL_STATS | CMD_CALL_NOWRAP;
     if (replicate) {
-        if (!(flags & REDISMODULE_ARGV_NO_AOF))
-            call_flags |= CMD_CALL_PROPAGATE_AOF;
-        if (!(flags & REDISMODULE_ARGV_NO_REPLICAS))
-            call_flags |= CMD_CALL_PROPAGATE_REPL;
+        if (replication_allowed) {
```

Comment:
i think it might have been cleaner (smaller diff) if we had turned off the `replicate` variable, rather than adding more `if`s and indentation (i may be wrong)